packages/dds/tree/src/feature-libraries/chunked-forest/codec/schemaBasedEncode.ts

```diff
@@ -122,7 +133,16 @@ export function treeShaper(
 
 		const objectNodeFields: KeyedFieldEncoder[] = [];
 		for (const [key, field] of schema.objectNodeFields ?? []) {
-			objectNodeFields.push({ key, encoder: fieldHandler.shapeFromField(field) });
+			// TODO: Remove this hardcoded check and do this based on either heuristic or schema.
+			const fieldEncoder =
+				shouldEncodeIncrementally &&
+				(key === "notes" || key === "label" || key === "labelText")
+					? new IncrementalFieldShape()
```

Comment:
For efficiency, its important that we do not make multiple equivalent shape instances: if you create a new shape instance every time, we can't deduplicate them and the encoded data will bloated.

To accomplish this, we produce all shapes using one of two patterns:
1. If the shape has no parameters, we only ever produce a single singleton instance.  NestedArrayShape.empty, AnyShape.instance
2. If the shape parameters, cache the shape as a function of its parameters in the EncoderCache (an implementation of the https://en.wikipedia.org/wiki/Flyweight_pattern)

Reply:
Updated this to be a singleton instance which also aligns with the other comments.